pallets/minterest-protocol/src/lib.rs

```diff
@@ -629,21 +611,18 @@ impl<T: Config> Pallet<T> {
 		);
 
 		// Fetch the amount the borrower owes, with accumulated interest
-		let account_borrows = <Controller<T>>::borrow_balance_stored(&borrower, underlying_asset_id)
-			.map_err(|_| Error::<T>::NumOverflow)?;
+		let account_borrows = <Controller<T>>::borrow_balance_stored(&borrower, underlying_asset_id)?;
 
-		repay_amount = match repay_amount.cmp(&Balance::zero()) {
-			Ordering::Equal => account_borrows,
-			_ => repay_amount,
-		};
+		if repay_amount.is_zero() {
+			repay_amount = account_borrows
+		}
 
 		ensure!(
 			repay_amount <= T::MultiCurrency::free_balance(underlying_asset_id, &who),
-			Error::<T>::NotEnoughUnderlyingsAssets
+			Error::<T>::NotEnoughUnderlyingAssets
```

Comment:
```suggestion
			Error::<T>::NotEnoughUnderlyingAsset
```
Assets mean several types of assets. While here have not enough of a specific asset type.

Reply:
Unfortunately I have no way to pass the asset type to the error.